amr-wind/ocean_waves/boundary_ops/OceanWavesBoundary.cpp

```diff
@@ -98,13 +110,26 @@ void OceanWavesBoundary::set_velocity(
             const auto& arr = mfab[mfi].array();
             const int numcomp = mfab.nComp();
 
+            const auto terrain_blank_flags =
+                terrain_and_vof ? (*m_terrain_blank_ptr)(lev).const_array(mfi)
+                                : amrex::Array4<int const>();
+
             amrex::ParallelFor(
                 bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
                     for (int n = 0; n < numcomp; n++) {
                         if (targ_vof(i, j, k) > constants::TIGHT_TOL) {
                             arr(i, j, k, dcomp + n) =
                                 targ_arr(i, j, k, orig_comp + n);
                         }
+                        if (terrain_and_vof) {
+                            // Terrain-blanked adjacent cell means 0 velocity
+                            const amrex::IntVect current_iv{i, j, k};
```

Comment:
you could just move this to the first line of the parallel for `const amrex::IntVect iv{i, j, k};` and replace all `(i,j,k,x)` by `(iv, x)`.